pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -163,6 +168,7 @@ protected void updateTopicPolicy(TopicPolicies data) {
                         data.getBackLogQuotaMap() == null ? null : data.getBackLogQuotaMap().get(type.toString())));
         topicPolicies.getTopicMaxMessageSize().updateTopicValue(data.getMaxMessageSize());
         topicPolicies.getMessageTTLInSeconds().updateTopicValue(data.getMessageTTLInSeconds());
+        topicPolicies.getSchemaCompatibilityStrategy().updateTopicValue(data.getSchemaCompatibilityStrategy());
```

Comment:
The update logic of SchemaCompatibilityStrategy is a bit different from other policies. SchemaCompatibilityStrategy.UNDEFINED has the same effect with null value, and System topic has special handling. Please refer to `setSchemaCompatibilityStrategy`, and make sure this change will keep the compatibility.